core/src/main/java/org/commonjava/indy/core/expire/ScheduleManagerBooter.java

```diff
@@ -26,14 +29,32 @@
 public class ScheduleManagerBooter
         implements BootupAction
 {
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private ScheduleDBConfig scheduleDBConfig;
+
     @Inject
-    private Instance<ScheduleManager> scheduleManager;
+    private Instance<ScheduleManager> scheduleManagers;
 
     @Override
     public void init()
             throws IndyLifecycleException
     {
-        scheduleManager.get().init();
+        for ( ScheduleManager scheduleManager : scheduleManagers )
```

Comment:
I'm not completely clear on when we would *not* init the schedule manager. It seems like if we're using the ScheduleDBManager but it's not enabled? Do we stack schedule managers somehow, so that we need to iterate through them?

Reply:
Okay, I prefer to implement the config for durable state https://projects.engineering.redhat.com/browse/MMENG-918 first and then improve this later. And it would be better to merge this after that config in place.

Reply:
Update the init based on the newly durable state config.